src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstanceEventsController.cs

```diff
@@ -0,0 +1,177 @@
+            .WriteTo.Console()
+            .CreateLogger();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InstanceEventsController"/> class
+        /// </summary>
+        /// <param name="instanceEventRepository">the instance repository handler</param>
+        public InstanceEventsController(IInstanceEventRepository instanceEventRepository)
+        {
+            _repository = instanceEventRepository;
+        }
+
+        /// <summary>
+        /// Inserts new instance event into the instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceEvent">Id of instance to retrieve events for. </param>
+        /// <returns>The stored instance event.</returns>
+        /// POST storage/api/v1/instanceevents/
+        [HttpPost]
+        public async Task<ActionResult> Post([FromBody] InstanceEvent instanceEvent)
+        {
+            if (instanceEvent == null)
+            {
+                return BadRequest("Missing parameter values: instance event must exist and instanceId must be set");
+            }
+
+            instanceEvent.CreatedDateTime = DateTime.UtcNow;
+
+            InstanceEvent result = await _repository.InsertInstanceEvent(instanceEvent);
+            if (result == null)
+            {
+                return BadRequest("Unable to write new instance event to database");
+            }
+
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for. </param>
+        /// <returns>List of instance events.</returns>
+        /// GET  storage/api/v1/instanceevents/?instanceid={instanceId}
+        [HttpGet]
+        public async Task<ActionResult> GetAllInstanceEvents(string instanceId)
+        {
+            if (!string.IsNullOrEmpty(instanceId))
+            {
+                List<InstanceEvent> result = await _repository.ListAllInstanceEvents(instanceId);
+                if (result == null || result.Count == 0)
+                {
+                    return NotFound($"Did not find any instance events for instanceId={instanceId}");
+                }
+
+                return Ok(result);
+            }
+
+            return BadRequest("Unable to perform query.");
+        }
+
+        /// <summary>
+        /// Retrieves all instance events related to given instance id and listed event types from instanceEvent collection.
+        /// </summary>
+        /// <param name="instanceId">Id of instance to retrieve events for.</param>
+        /// <param name="eventTypes">List of event types to filter the events by./param>
+        /// <returns>List of intance events.</returns>
+        /// GET  storage/api/v1/instanceevents/GetByInstanceEventType?instanceId={instanceId}&eventTypes={eventType}&eventTypes={eventType}
```

Comment:
Route should not include method name (GetByInstanceEventType)
the url should be api/v1/instanceevents?instanceId={instanceId}&eventTypes={eventType}
This applies to most of the methods here

Reply:
@hannekot Implementing it that way caused the routes to be ambigous and the API requests to fail.. Is there a way I can ensure that specific parameter combinations are routed to a specific function ?

Reply:
Updated route to storage/api/v1/instances/{instanceId}/events